controller/volume_controller.go

```diff
@@ -242,7 +242,14 @@ func (vc *VolumeController) syncVolume(key string) (err error) {
 	volume, err := vc.ds.GetVolume(name)
 	if err != nil {
 		if datastore.ErrorIsNotFound(err) {
-			vc.logger.WithField("volume", name).Debug("Longhorn volume has been deleted")
+			log := vc.logger.WithField("volume", name)
+			log.Debug("Longhorn volume has been deleted")
+
+			if sm, _ := vc.ds.GetShareManager(name); sm != nil {
+				log.Info("Removing orphaned share manager for deleted volume")
+				_ = vc.ds.DeleteShareManager(name)
```

Comment:
Failure to delete should expose a warning.

Reply:
Removed, since I added a volume owner reference to the share manager so kubernetes can garbage collect them.
If we miss the deletion for whatever reason.